### PR TITLE
fix ddp

### PR DESCRIPTION
**Pull request by kuba-krj**
# Description
Two changes were needed to make DDP work again:
1. Add `if self.logger is not None` in logging time (only process on gpu 0 has a Neptune logger)
2. Change `gpus={N_GPUS}` to `gres` option in sbatch/slurm. The latter will only allocate resources on one node. In our current codebase we don't support multi-node training.

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer:yes
4. Are all variables set with logical defaults that are backward compatible?
   - Answer:yes
5. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   -  Answer: [Link to Neptune](https://app.neptune.ai/o/pmtest/org/llm-efficiency/runs/details?viewId=98a55dc0-aa23-4c1a-a622-93ec989d9ab3&detailsTab=charts&shortId=LLMEF-5803)
6. Have you successfully executed the linter and tests?
   - Answer:yes
7. Are all functions concise and don't require splitting?
   - Answer:yes
8. Is there documentation for arguments and complex logic?
   - Answer:yes
9. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer:yes
10. Are all functions placed in appropriate files?
   - Answer:yes
11. Is the PR name meaningful and descriptive?
   - Answer:yes
12. Is PR description provided, or unnecessary?
    - Answer:yes
  
Other questions:

13. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer:yes
14. Is the change significant enough to notify all individuals working with this repository?
    - Answer:yes
15. Is there a need for a second reviewer?
    - Answer:yes

### CODE REVIEW

**Review by sebastianjaszczur — APPROVED** [review]
LGTM, ale poczekałbym na innych reviewerów.

File: research/conditional/utils/conditional_trainer.py [hunk 1]
```diff
@@ -134,7 +134,11 @@ def train(self, n_steps: int):
             if step % 1000 == 0:
                 print(f"Step {step}")
 
-            if self.model_type == "gpt" and step % self.decoding_logging_steps == 0:
+            if (
+                self.model_type == "gpt"
+                and step % self.decoding_logging_steps == 0
+                and self.logger is not None
```
Comment (crewtool) [inline research/conditional/utils/conditional_trainer.py:140]: Może dodajmy komentarz, że tylko cuda:0 ma loggera, bo to nieoczywiste

File: research/conditional/utils/conditional_trainer.py [hunk 2]
```diff
@@ -146,7 +150,7 @@ def train(self, n_steps: int):
             self.total_time_decoding += t2 - t1
             self.total_time_afterstep += t3 - t2
 
-            if step % 1000 == 0:
+            if step % 1000 == 0 and self.logger is not None:
```
Comment (crewtool) [inline research/conditional/utils/conditional_trainer.py:153]: Same

**Review by crewtool — APPROVED** [review]

File: research/conditional/utils/conditional_trainer.py (hunk 1 above)
Comment (maciek-pioro) [reply research/conditional/utils/conditional_trainer.py:140]: @crewtool  @kuba-krj a może dałoby się dodać parametr w rodzaju `self.is_logging_process`?

**Review by maciek-pioro — COMMENTED** [review]
Dodałem komentarz w podobnym duchu co @crewtool , rozważmy jak to ładnie zrobić

File: research/conditional/utils/conditional_trainer.py (hunk 2 above)
Comment (maciek-pioro) [reply research/conditional/utils/conditional_trainer.py:153]: jw

File: research/conditional/utils/conditional_trainer.py (hunk 1 above)
Comment (kuba-krj) [reply research/conditional/utils/conditional_trainer.py:140]: Dodałem parametr

File: research/conditional/utils/conditional_trainer.py (hunk 2 above)
Comment (kuba-krj) [reply research/conditional/utils/conditional_trainer.py:153]: Dodałem parametr

**Review by crewtool — APPROVED** [review]
LGTM!

**Pull request merged**